lib/wallaroo/core/source/connector_source/connector_framed_source_notify.pony

```diff
@@ -115,9 +132,16 @@ class ConnectorSourceNotify[In: Any val]
           // TODO: We need a way to determine the key based on the policy
           // for any particular connector. For example, the Kafka connector
           // needs a way to provide the Kafka key here.
-          let initial_key = msg_uid.string()
 
-          if decoded isnt None then
+          let initial_key =
+            if key_string isnt None then
+              key_string.string()
```

Comment:
since `key_string` is already a `String` type from above, the `.string()` call can be ommitted here.